netsuitesdk/internal/client.py

```diff
@@ -533,34 +533,38 @@ def basic_stringfield_search(self, type_name, attribute, value, operator=None):
         if result.records:
             return result.records
 
-    # def upsertList(self, records):
-    #     """
-    #     Add objects of type recordType with given externalId..
-    #     If a record of specified type with matching externalId already
-    #     exists, it is updated.
-
-    #     Usage example:
-    #         customer1 = self.Customer(externalId='customer', email='test1@example.com')
-    #         customer2 = self.Customer(externalId='another_customer', email='test2@example.com')
-    #         self.upsertList(records=[customer1, customer2])
-
-    #     :param list[CompoundValue] records: the records to be created or updated
-    #     :return: a reference to the newly created or updated records
-    #     :rtype: list[CompoundValue]
-    #     """
-
-    #     response = self.request('upsertList', record=records)
-    #     responses = response.body.writeResponse
-    #     record_refs = []
-    #     for response in responses:
-    #         status = response.status
-    #         if status.isSuccess:
-    #             record_ref = response['baseRef']
-    #             self.logger.debug('Successfully updated record of type {type}, internalId: {internalId}, externalId: {externalId}'.format(
-    #                     type=record_ref['type'], internalId=record_ref['internalId'], externalId=record_ref['externalId']))
-    #             record_refs.append(record_ref)
-    #         else:
-    #             exc = self._request_error('upsertList', detail=status['statusDetail'][0])
-    #             has_failures = True
-    #             raise exc
-    #     return record_refs
+    def upsert_list(self, records):
+        """
+        Add objects of type recordType with given externalId..
+        If a record of specified type with matching externalId already
+        exists, it is updated.
+
+        Usage example:
+            customer1 = self.Customer(externalId='customer', email='test1@example.com')
+            customer2 = self.Customer(externalId='another_customer', email='test2@example.com')
+            self.upsert_list(records=[customer1, customer2])
+
+        :param list[CompoundValue] records: the records to be created or updated
+        :return: a reference to the newly created or updated records, with a success/error status
+        :rtype: list[OrderedDict]
+        """
+
+        response = self.request('upsertList', record=records)
+        responses = response.body.writeResponseList
+        status = responses['status']
+        record_refs = []
+        if status.isSuccess:
+            for response in responses.writeResponse:
+                object_reference = response['baseRef']
+                record_ref = {
+                    "object_reference": object_reference,
+                    "status": response["status"]
+                }
+                self.logger.debug('Successfully updated record of type {type}, internalId: {internalId}, externalId: {externalId}'.format(
```

Comment:
will this log to datadog even though it's in this package?

Reply:
no, this is just their internal logger, and because it's debug, it's probably just a development aid